pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -38,6 +43,59 @@ func (e *contentRemainingError) Error() string {
 	return fmt.Sprintf("some content remains in the namespace, estimate %d seconds before it is removed", e.Estimate)
 }
 
+// operation is used for caching if an operation is supported on a dynamic client.
+type operation string
+
+const (
+	operationDeleteCollection operation = "deleteCollection"
+	operationList             operation = "list"
+)
+
+// operationSupportedCacheEntry is an entry in the cache.
+type operationSupportedCacheEntry struct {
+	op                   operation
+	groupVersionResource unversioned.GroupVersionResource
+	supported            bool
+}
+
+// operationSupportedCache is a simple cache to remember if an operation is supported for a resource.
+type operationSupportedCache struct {
+	store cache.Store
+}
```

Comment:
i think i was unncessarily worried about locking late at night when i did this, when i don't think that is a major concern...  i will look to update.